### PR TITLE
feat: support cx355001 fan

### LINKED ISSUE #137: Fan CX3550/01 Missing
Hi,
is it possible to add the New Philips WLAN Fan cx3550?
It can be controled by the Philips Air+ App, as well as the  Philips Purifier AC2889.

Greetings 
BA2000

### PR DESCRIPTION
**Pull request by EuleMitKeule**
This PR adds support for the new CX3550/01 fan.
All features are working, including power cycling, speed mode selection, oscillation and timer.

Closes #137 

Credit also goes to @martlxmartl, @werty1st and @wehrmannit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Note: `configuration.yaml` is no longer supported and your configuration is not
 - AMF765
 - AMF870
 - CX5120
+- CX3550/01
```
Comment (kongo09) [inline README.md:92]: You should be able to leave out the `/01` as this is (most of the time) only a model variant by Philips referring to color of the casing or region of sales. Ideally, all CX3550 models should work and the integration will pick that up.

File: custom_components/philips_airpurifier_coap/const.py [hunk 2]
```diff
@@ -127,6 +127,7 @@ class FanModel(StrEnum):
     AMF765 = "AMF765"
     AMF870 = "AMF870"
     CX5120 = "CX5120"
+    CX3550 = "CX3550/01"
```
Comment (kongo09) [inline custom_components/philips_airpurifier_coap/const.py:130]: please leavel out the `/01` as discussed above

File: custom_components/philips_airpurifier_coap/philips.py [hunk 3]
```diff
@@ -1785,6 +1785,91 @@ class PhilipsCX5120(PhilipsNew2GenericCoAPFan):
...
+    }
+    KEY_OSCILLATION = {
+        PhilipsApi.NEW2_OSCILLATION: PhilipsApi.OSCILLATION_MAP2,
+    }
+
+    AVAILABLE_SWITCHES = [PhilipsApi.NEW2_BEEP]
+    AVAILABLE_SELECTS = [PhilipsApi.NEW2_TIMER2]
+
+    @property
+    def oscillating(self) -> bool | None:
```
Comment (kongo09) [inline custom_components/philips_airpurifier_coap/philips.py:1851]: This property is already part of `PhilipsGenericCoAPFanBase` with the difference of the `on` test. I suggest you put the special test into that location. It's all not perfect, but I rather avoid the code duplication.

File: custom_components/philips_airpurifier_coap/philips.py [hunk 4]
```diff
@@ -1785,6 +1785,91 @@ class PhilipsCX5120(PhilipsNew2GenericCoAPFan):
...
+        """Return if the fan is oscillating."""
+
+        key = next(iter(self.KEY_OSCILLATION))
+        status = self._device_status.get(key)
+        on = self.KEY_OSCILLATION.get(key).get(SWITCH_ON)
+        return status is not None and status in [on, PhilipsApi.OSCILLATION_CX355001]
+
+    async def async_oscillate(self, oscillating: bool) -> None:
```
Comment (kongo09) [inline custom_components/philips_airpurifier_coap/philips.py:1859]: Again, I'd prefer if we kept the code in one place and the class is only used to configure. I see that you did a bit more work here. Do you think you can merge that with the existing code in `PhilipsGenericCoAPFanBase`?

**Comment by kongo09** [comment]
Thanks a lot for your great work. If you could help out with the suggested adaptations above, I can merge it in. It's great to have others contribute and take some of the load!

**Comment by EuleMitKeule** [comment]
I am not able to see any suggestions. Also, I believe there are still some bugs, therefore I will convert this PR to a draft for now.
grafik

**Comment by kongo09** [comment]
Don't you see the comments I've left?

**Review by kongo09 — CHANGES_REQUESTED** [review]
Sorry, I forgot to submit the review

**Comment by kongo09** [comment]
@EuleMitKeule will you continue to work on this?

**Comment by EuleMitKeule** [comment]
I might, got not much time currently.

**Comment by kongo09** [comment]
I know that problem ;-)

I also noticed that you did some bigger changes in your fork. Looks interesting but certainly requires a bit of thinking to pull that over. Thanks for engaging with the code that much and helping out!

Just let me know when you find more time - I might pull in your code for the CX model in the meantime.

**Comment by EuleMitKeule** [comment]
Yeah, I had some problem getting the integration to work at all, that's why I started to make some larger changes to the code. Controlling the fan in the fork is working for me, but displaying the state is still bugged.

**Comment by kongo09** [comment]
I've done above changes after merging.

**Pull request merged**